libraries/AP_HAL_ChibiOS/RCOutput.cpp

```diff
@@ -253,9 +255,11 @@ void RCOutput::dshot_collect_dma_locks(uint32_t time_out_us)
             // if we have time left wait for the event
             eventmask_t mask = 0;
             const uint32_t pulse_elapsed_us = now - group.last_dmar_send_us;
-            if (now < time_out_us) {
+            // will be >> timeout_period_us if timeout_us is in the past
+            const uint32_t timeout_remaining_us = timeout_us - now;
+            if (timeout_remaining_us > 0 && timeout_remaining_us <= timeout_period_us) { // wrap sensitive
                 mask = chEvtWaitOneTimeout(group.dshot_event_mask,
-                    chTimeUS2I(MAX(time_out_us - now, group.dshot_pulse_send_time_us - pulse_elapsed_us)));
+                    chTimeUS2I(MAX(timeout_us - now, group.dshot_pulse_send_time_us - pulse_elapsed_us)));
```

Comment:
I'd like a MIN() with a known small number to protect against any possibility of wrap. 
same goes for the one in the next if()

Reply:
Ok added